apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+    * @param _challengeId Current challenge identification number for the action
+    * @param _challenge Current challenge instance for the action
+    */
+    function _voidChallenge(uint256 _actionId, Action storage _action, uint256 _challengeId, Challenge storage _challenge) internal {
+        _challenge.state = ChallengeState.Voided;
+
+        (IDisputable disputable, CollateralRequirement storage requirement) = _getDisputableFor(_action);
+        _transfer(requirement.token, _challenge.challenger, requirement.challengeAmount);
+        disputable.onDisputableActionVoided(_action.disputableActionId);
+        emit ActionVoided(_actionId, _challengeId);
+    }
+
+    /**
+    * @dev Lock some tokens in the staking pool for a user
+    * @param _staking Staking pool for the ERC20 token to be locked
+    * @param _user Address of the user to lock tokens for
+    * @param _amount Number of collateral tokens to be locked
+    */
+    function _lockBalance(Staking _staking, address _user, uint256 _amount) internal {
+        if (_amount == 0) {
+            return;
+        }
+
+        _staking.lock(_user, _amount);
+    }
+
+    /**
+    * @dev Unlock a number of tokens in the staking pool for a user
+    * @param _staking Staking pool for the ERC20 token to be unlocked
+    * @param _user Address of the user to unlock tokens for
+    * @param _amount Number of collateral tokens to be unlocked
+    */
+    function _unlockBalance(Staking _staking, address _user, uint256 _amount) internal {
+        if (_amount == 0) {
+            return;
+        }
+
+        _staking.unlock(_user, _amount);
+    }
+
+    /**
+    * @dev Slash a number of tokens in the staking pool from a user to a challenger
+    * @param _staking Staking pool for the ERC20 token to be slashed
+    * @param _user Address of the user to be slashed
+    * @param _challenger Address receiving the slashed tokens
+    * @param _amount Number of collateral tokens to be slashed
+    */
+    function _slashBalance(Staking _staking, address _user, address _challenger, uint256 _amount) internal {
+        if (_amount == 0) {
+            return;
+        }
+
+        _staking.slash(_user, _challenger, _amount);
+    }
+
+    /**
+    * @dev Unlock and slash a number of tokens in the staking pool from a user in favour of a challenger
+    * @param _staking Staking pool for the ERC20 token to be unlocked and slashed
+    * @param _user Address of the user to be unlocked and slashed
+    * @param _unlockAmount Number of collateral tokens to be unlocked
+    * @param _challenger Address receiving the slashed tokens
+    * @param _slashAmount Number of collateral tokens to be slashed
+    */
+    function _unlockAndSlashBalance(Staking _staking, address _user, uint256 _unlockAmount, address _challenger, uint256 _slashAmount) internal {
+        if (_unlockAmount != 0 && _slashAmount != 0) {
+            _staking.unlockAndSlash(_user, _unlockAmount, _challenger, _slashAmount);
```

Comment:
⚠️ Maybe this is addressed in the staking PR, but I would assume the staking module to handle 0 for either of the amounts, so we should just that check that one of the amounts is non-zero.